File: Week 03/id_701/LeetCode_102_701.py
# Definition for a binary tree node.
class TreeNode:
    def __init__(self, x):
        self.val = x
        self.left = None
        self.right = None


class Solution:
    def levelOrder(self, root: TreeNode) -> list:
        levels = []
        if not root:
            return levels

        def helper(node: TreeNode, level: int):
            if level(levels) == level:
                levels.append([])
                
            levels[level].append(node.val)

            if root.left:
                helper(root.left, level + 1)
            if root.right:
                helper(root.right, level+1)

        helper(root, 0)
        return levels

File: Week 03/id_701/LeetCode_102_701.cs
/**
 * Definition for a binary tree node. */
 public class TreeNode {
      public int val;
      public TreeNode left;
      public TreeNode right;
      public TreeNode(int x) { val = x; }
  }

public class Solution {
    public IList<IList<int>> LevelOrder(TreeNode root) {
        var levels = new List<IList<int>>();
        if(root == null){
            return levels;
        }
        
        void Helper(TreeNode node, int level){
            if(levels.Count == level){
                levels.Add(new List<int>());
            }
            levels[level].Add(node.val);
            
            if(node.left != null){
                Helper(node.left, level + 1);
            }
            if(node.right != null){
                Helper(node.right, level + 1);
            }
        }
        
        Helper(root, 0);
        return levels;
    }
}